schema.graphql

```diff
@@ -4,13 +4,25 @@ enum StakeType {
   T
 }
 
+type Account @entity {
+  # Id is the ethereum address of the account.
+  id: ID!
+  stakes: [StakeData!] @derivedFrom(field: "owner")
+  delegatee: TokenholderDelegation
+  # TODO add more fields example `tokenBalances`.
+}
+
 type StakeData @entity {
   id: ID! # staking provider address
+  owner: Account! # address
   stakeType: StakeType!
```

Comment:
But stakes don't have a single type. What they have is 3 different T sources, which can occur at the same time, and that, combined, make up for the total stake amount. Another thing is that, in practice, many stakes will only have one T source, but there are also many cases where they have two (e.g., a legacy KEEP staker migrates and later tops up with T; a staker that creates a T stake and later tops up using a legacy NU stake, etc).

I understand the dapp need to discern if a given stake comes from a legacy contract, but the way to do it should be checking if either `nuInTStake` or `keepInTStake` are > 0. An alternative, would be to create a boolean `isLegacy` that's populated depending on the changes to `nuInTStake` or `keepInTStake`. Another alternative, is to allow `stakeType` to be multi-value somehow, but I think that's overkill.

Reply:
> We keep the stake type on-chain so that's why I'm ok with setting the stakeType in subgraph.

Well, but we have the stake type only for events, so they only have "historical" value. As you rightfully sat, someone can create a legacy stake and, with time, end up only with T tokens. Do we need the dapp to be aware this was the case? IMO, no: we only care that a stake is _currently_ related to a legacy stake. 

In any case, this is not related to your PR, and we can decide in a different moment if we keep the `stakeType` or not.

Reply:
IMHO this field is not necessary and it was added from the erroneous assumption that stakes only have a single type.

But I'm with @cygnusv. We can change this later.